main.go

```diff
@@ -171,8 +174,33 @@ func main() {
 		w.Write([]byte("Only POST or PUT requests allowed."))
 	}))
 
+	mux := http.NewServeMux()
+	mux.Handle("/", r)
+
+	buildTime := time.Now()
+	buildInfo := map[string]string{
+		"version":   version.Version,
+		"revision":  version.Revision,
+		"branch":    version.Branch,
+		"buildUser": version.BuildUser,
+		"buildDate": version.BuildDate,
+		"goVersion": version.GoVersion,
+	}
+
+	apiV1 := api_V1.New(logger, ms, flags, buildTime, buildInfo)
+
+	apiPath := "/api"
+	if *routePrefix != "/" {
+		apiPath = *routePrefix + apiPath
+		level.Info(logger).Log("msg", "router prefix", "prefix", *routePrefix)
+	}
+	av1 := route.New()
+	apiV1.Register(av1)
+
+	mux.Handle(apiPath+"/v1/", http.StripPrefix(apiPath+"/v1", av1))
```

Comment:
Hmm, so now we are mixing up direct use of `httprouter` with the use through `prometheus/common/route`. Perhaps that's OK. But we still need to solve instrumentation for the API (which I assume to be easy). Perhaps add TODOs for both (instrument the endpoints served via `prometheus/common/route` as well as unifying the usage).